// @letta-ai/letta-client/serialization/resources/templates/types/TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItem.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItem = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLimit_1 = require("./TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLimit");
const TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemName_1 = require("./TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemName");
const TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemIsTemplate_1 = require("./TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemIsTemplate");
const TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLabel_1 = require("./TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLabel");
const TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemDescription_1 = require("./TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemDescription");
const TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemMetadata_1 = require("./TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemMetadata");
const TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemId_1 = require("./TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemId");
const TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemOrganizationId_1 = require("./TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemOrganizationId");
const TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemCreatedById_1 = require("./TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemCreatedById");
const TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLastUpdatedById_1 = require("./TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLastUpdatedById");
exports.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItem = core.serialization.object({
    value: core.serialization.string(),
    limit: TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLimit_1.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLimit.optional(),
    name: TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemName_1.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemName.optional(),
    isTemplate: core.serialization.property("is_template", TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemIsTemplate_1.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemIsTemplate.optional()),
    label: TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLabel_1.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLabel.optional(),
    description: TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemDescription_1.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemDescription.optional(),
    metadata: TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemMetadata_1.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemMetadata.optional(),
    id: TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemId_1.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemId.optional(),
    organizationId: core.serialization.property("organization_id", TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemOrganizationId_1.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemOrganizationId.optional()),
    createdById: core.serialization.property("created_by_id", TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemCreatedById_1.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemCreatedById.optional()),
    lastUpdatedById: core.serialization.property("last_updated_by_id", TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLastUpdatedById_1.TemplatesCreateAgentsResponseAgentsItemMemoryBlocksItemLastUpdatedById.optional()),
});
