// @letta-ai/letta-client/serialization/resources/templates/types/TemplatesCreateAgentsResponseAgentsItemSourcesItem.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsResponseAgentsItemSourcesItem = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsResponseAgentsItemSourcesItemId_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemId");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemDescription_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemDescription");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfig_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfig");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemOrganizationId_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemOrganizationId");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemMetadata_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemMetadata");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemCreatedById_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemCreatedById");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemLastUpdatedById_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemLastUpdatedById");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemCreatedAt_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemCreatedAt");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemUpdatedAt_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemUpdatedAt");
exports.TemplatesCreateAgentsResponseAgentsItemSourcesItem = core.serialization.object({
    id: TemplatesCreateAgentsResponseAgentsItemSourcesItemId_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemId.optional(),
    name: core.serialization.string(),
    description: TemplatesCreateAgentsResponseAgentsItemSourcesItemDescription_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemDescription.optional(),
    embeddingConfig: core.serialization.property("embedding_config", TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfig_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfig),
    organizationId: core.serialization.property("organization_id", TemplatesCreateAgentsResponseAgentsItemSourcesItemOrganizationId_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemOrganizationId.optional()),
    metadata: TemplatesCreateAgentsResponseAgentsItemSourcesItemMetadata_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemMetadata.optional(),
    createdById: core.serialization.property("created_by_id", TemplatesCreateAgentsResponseAgentsItemSourcesItemCreatedById_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemCreatedById.optional()),
    lastUpdatedById: core.serialization.property("last_updated_by_id", TemplatesCreateAgentsResponseAgentsItemSourcesItemLastUpdatedById_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemLastUpdatedById.optional()),
    createdAt: core.serialization.property("created_at", TemplatesCreateAgentsResponseAgentsItemSourcesItemCreatedAt_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemCreatedAt.optional()),
    updatedAt: core.serialization.property("updated_at", TemplatesCreateAgentsResponseAgentsItemSourcesItemUpdatedAt_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemUpdatedAt.optional()),
});
