// @letta-ai/letta-client/serialization/resources/templates/types/TemplatesCreateAgentsResponseAgentsItem.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsResponseAgentsItem = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsResponseAgentsItemCreatedById_1 = require("./TemplatesCreateAgentsResponseAgentsItemCreatedById");
const TemplatesCreateAgentsResponseAgentsItemLastUpdatedById_1 = require("./TemplatesCreateAgentsResponseAgentsItemLastUpdatedById");
const TemplatesCreateAgentsResponseAgentsItemCreatedAt_1 = require("./TemplatesCreateAgentsResponseAgentsItemCreatedAt");
const TemplatesCreateAgentsResponseAgentsItemUpdatedAt_1 = require("./TemplatesCreateAgentsResponseAgentsItemUpdatedAt");
const TemplatesCreateAgentsResponseAgentsItemToolRules_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolRules");
const TemplatesCreateAgentsResponseAgentsItemMessageIds_1 = require("./TemplatesCreateAgentsResponseAgentsItemMessageIds");
const TemplatesCreateAgentsResponseAgentsItemAgentType_1 = require("./TemplatesCreateAgentsResponseAgentsItemAgentType");
const TemplatesCreateAgentsResponseAgentsItemLlmConfig_1 = require("./TemplatesCreateAgentsResponseAgentsItemLlmConfig");
const TemplatesCreateAgentsResponseAgentsItemEmbeddingConfig_1 = require("./TemplatesCreateAgentsResponseAgentsItemEmbeddingConfig");
const TemplatesCreateAgentsResponseAgentsItemOrganizationId_1 = require("./TemplatesCreateAgentsResponseAgentsItemOrganizationId");
const TemplatesCreateAgentsResponseAgentsItemDescription_1 = require("./TemplatesCreateAgentsResponseAgentsItemDescription");
const TemplatesCreateAgentsResponseAgentsItemMetadata_1 = require("./TemplatesCreateAgentsResponseAgentsItemMetadata");
const TemplatesCreateAgentsResponseAgentsItemMemory_1 = require("./TemplatesCreateAgentsResponseAgentsItemMemory");
const TemplatesCreateAgentsResponseAgentsItemToolsItem_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolsItem");
const TemplatesCreateAgentsResponseAgentsItemSourcesItem_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItem");
const TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariables_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariables");
const TemplatesCreateAgentsResponseAgentsItemProjectId_1 = require("./TemplatesCreateAgentsResponseAgentsItemProjectId");
const TemplatesCreateAgentsResponseAgentsItemTemplateId_1 = require("./TemplatesCreateAgentsResponseAgentsItemTemplateId");
const TemplatesCreateAgentsResponseAgentsItemBaseTemplateId_1 = require("./TemplatesCreateAgentsResponseAgentsItemBaseTemplateId");
const TemplatesCreateAgentsResponseAgentsItemIdentityIds_1 = require("./TemplatesCreateAgentsResponseAgentsItemIdentityIds");
const TemplatesCreateAgentsResponseAgentsItemMessageBufferAutoclear_1 = require("./TemplatesCreateAgentsResponseAgentsItemMessageBufferAutoclear");
exports.TemplatesCreateAgentsResponseAgentsItem = core.serialization.object({
    createdById: core.serialization.property("created_by_id", TemplatesCreateAgentsResponseAgentsItemCreatedById_1.TemplatesCreateAgentsResponseAgentsItemCreatedById.optional()),
    lastUpdatedById: core.serialization.property("last_updated_by_id", TemplatesCreateAgentsResponseAgentsItemLastUpdatedById_1.TemplatesCreateAgentsResponseAgentsItemLastUpdatedById.optional()),
    createdAt: core.serialization.property("created_at", TemplatesCreateAgentsResponseAgentsItemCreatedAt_1.TemplatesCreateAgentsResponseAgentsItemCreatedAt.optional()),
    updatedAt: core.serialization.property("updated_at", TemplatesCreateAgentsResponseAgentsItemUpdatedAt_1.TemplatesCreateAgentsResponseAgentsItemUpdatedAt.optional()),
    id: core.serialization.string(),
    name: core.serialization.string(),
    toolRules: core.serialization.property("tool_rules", TemplatesCreateAgentsResponseAgentsItemToolRules_1.TemplatesCreateAgentsResponseAgentsItemToolRules.optional()),
    messageIds: core.serialization.property("message_ids", TemplatesCreateAgentsResponseAgentsItemMessageIds_1.TemplatesCreateAgentsResponseAgentsItemMessageIds.optional()),
    system: core.serialization.string(),
    agentType: core.serialization.property("agent_type", TemplatesCreateAgentsResponseAgentsItemAgentType_1.TemplatesCreateAgentsResponseAgentsItemAgentType),
    llmConfig: core.serialization.property("llm_config", TemplatesCreateAgentsResponseAgentsItemLlmConfig_1.TemplatesCreateAgentsResponseAgentsItemLlmConfig),
    embeddingConfig: core.serialization.property("embedding_config", TemplatesCreateAgentsResponseAgentsItemEmbeddingConfig_1.TemplatesCreateAgentsResponseAgentsItemEmbeddingConfig),
    organizationId: core.serialization.property("organization_id", TemplatesCreateAgentsResponseAgentsItemOrganizationId_1.TemplatesCreateAgentsResponseAgentsItemOrganizationId.optional()),
    description: TemplatesCreateAgentsResponseAgentsItemDescription_1.TemplatesCreateAgentsResponseAgentsItemDescription.optional(),
    metadata: TemplatesCreateAgentsResponseAgentsItemMetadata_1.TemplatesCreateAgentsResponseAgentsItemMetadata.optional(),
    memory: TemplatesCreateAgentsResponseAgentsItemMemory_1.TemplatesCreateAgentsResponseAgentsItemMemory,
    tools: core.serialization.list(TemplatesCreateAgentsResponseAgentsItemToolsItem_1.TemplatesCreateAgentsResponseAgentsItemToolsItem),
    sources: core.serialization.list(TemplatesCreateAgentsResponseAgentsItemSourcesItem_1.TemplatesCreateAgentsResponseAgentsItemSourcesItem),
    tags: core.serialization.list(core.serialization.string()),
    toolExecEnvironmentVariables: core.serialization.property("tool_exec_environment_variables", TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariables_1.TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariables.optional()),
    projectId: core.serialization.property("project_id", TemplatesCreateAgentsResponseAgentsItemProjectId_1.TemplatesCreateAgentsResponseAgentsItemProjectId.optional()),
    templateId: core.serialization.property("template_id", TemplatesCreateAgentsResponseAgentsItemTemplateId_1.TemplatesCreateAgentsResponseAgentsItemTemplateId.optional()),
    baseTemplateId: core.serialization.property("base_template_id", TemplatesCreateAgentsResponseAgentsItemBaseTemplateId_1.TemplatesCreateAgentsResponseAgentsItemBaseTemplateId.optional()),
    identityIds: core.serialization.property("identity_ids", TemplatesCreateAgentsResponseAgentsItemIdentityIds_1.TemplatesCreateAgentsResponseAgentsItemIdentityIds.optional()),
    messageBufferAutoclear: core.serialization.property("message_buffer_autoclear", TemplatesCreateAgentsResponseAgentsItemMessageBufferAutoclear_1.TemplatesCreateAgentsResponseAgentsItemMessageBufferAutoclear.optional()),
});
