// @letta-ai/letta-client/serialization/resources/templates/types/TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfig.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfig = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpointType_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpointType");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpoint_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpoint");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingChunkSize_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingChunkSize");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigHandle_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigHandle");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureEndpoint_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureEndpoint");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureVersion_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureVersion");
const TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureDeployment_1 = require("./TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureDeployment");
exports.TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfig = core.serialization.object({
    embeddingEndpointType: core.serialization.property("embedding_endpoint_type", TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpointType_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpointType),
    embeddingEndpoint: core.serialization.property("embedding_endpoint", TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpoint_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpoint.optional()),
    embeddingModel: core.serialization.property("embedding_model", core.serialization.string()),
    embeddingDim: core.serialization.property("embedding_dim", core.serialization.number()),
    embeddingChunkSize: core.serialization.property("embedding_chunk_size", TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingChunkSize_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingChunkSize.optional()),
    handle: TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigHandle_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigHandle.optional(),
    azureEndpoint: core.serialization.property("azure_endpoint", TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureEndpoint_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureEndpoint.optional()),
    azureVersion: core.serialization.property("azure_version", TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureVersion_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureVersion.optional()),
    azureDeployment: core.serialization.property("azure_deployment", TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureDeployment_1.TemplatesCreateAgentsResponseAgentsItemSourcesItemEmbeddingConfigAzureDeployment.optional()),
});
