// @letta-ai/letta-client/serialization/resources/templates/types/TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItem.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItem = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemCreatedById_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemCreatedById");
const TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemLastUpdatedById_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemLastUpdatedById");
const TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemCreatedAt_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemCreatedAt");
const TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemUpdatedAt_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemUpdatedAt");
const TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemId_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemId");
const TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemDescription_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemDescription");
const TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemOrganizationId_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemOrganizationId");
exports.TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItem = core.serialization.object({
    createdById: core.serialization.property("created_by_id", TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemCreatedById_1.TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemCreatedById.optional()),
    lastUpdatedById: core.serialization.property("last_updated_by_id", TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemLastUpdatedById_1.TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemLastUpdatedById.optional()),
    createdAt: core.serialization.property("created_at", TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemCreatedAt_1.TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemCreatedAt.optional()),
    updatedAt: core.serialization.property("updated_at", TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemUpdatedAt_1.TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemUpdatedAt.optional()),
    id: TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemId_1.TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemId.optional(),
    key: core.serialization.string(),
    value: core.serialization.string(),
    description: TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemDescription_1.TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemDescription.optional(),
    organizationId: core.serialization.property("organization_id", TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemOrganizationId_1.TemplatesCreateAgentsResponseAgentsItemToolExecEnvironmentVariablesItemOrganizationId.optional()),
    agentId: core.serialization.property("agent_id", core.serialization.string()),
});
