// @letta-ai/letta-client/Client.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LettaClient = void 0;
const Client_1 = require("./api/resources/tools/client/Client");
const Client_2 = require("./api/resources/sources/client/Client");
const Client_3 = require("./api/resources/agents/client/Client");
const Client_4 = require("./api/resources/identities/client/Client");
const Client_5 = require("./api/resources/models/client/Client");
const Client_6 = require("./api/resources/blocks/client/Client");
const Client_7 = require("./api/resources/jobs/client/Client");
const Client_8 = require("./api/resources/health/client/Client");
const Client_9 = require("./api/resources/providers/client/Client");
const Client_10 = require("./api/resources/runs/client/Client");
const Client_11 = require("./api/resources/steps/client/Client");
const Client_12 = require("./api/resources/tag/client/Client");
const Client_13 = require("./api/resources/voice/client/Client");
const Client_14 = require("./api/resources/templates/client/Client");
class LettaClient {
    constructor(_options = {}) {
        this._options = _options;
    }
    get tools() {
        var _a;
        return ((_a = this._tools) !== null && _a !== void 0 ? _a : (this._tools = new Client_1.Tools(this._options)));
    }
    get sources() {
        var _a;
        return ((_a = this._sources) !== null && _a !== void 0 ? _a : (this._sources = new Client_2.Sources(this._options)));
    }
    get agents() {
        var _a;
        return ((_a = this._agents) !== null && _a !== void 0 ? _a : (this._agents = new Client_3.Agents(this._options)));
    }
    get identities() {
        var _a;
        return ((_a = this._identities) !== null && _a !== void 0 ? _a : (this._identities = new Client_4.Identities(this._options)));
    }
    get models() {
        var _a;
        return ((_a = this._models) !== null && _a !== void 0 ? _a : (this._models = new Client_5.Models(this._options)));
    }
    get blocks() {
        var _a;
        return ((_a = this._blocks) !== null && _a !== void 0 ? _a : (this._blocks = new Client_6.Blocks(this._options)));
    }
    get jobs() {
        var _a;
        return ((_a = this._jobs) !== null && _a !== void 0 ? _a : (this._jobs = new Client_7.Jobs(this._options)));
    }
    get health() {
        var _a;
        return ((_a = this._health) !== null && _a !== void 0 ? _a : (this._health = new Client_8.Health(this._options)));
    }
    get providers() {
        var _a;
        return ((_a = this._providers) !== null && _a !== void 0 ? _a : (this._providers = new Client_9.Providers(this._options)));
    }
    get runs() {
        var _a;
        return ((_a = this._runs) !== null && _a !== void 0 ? _a : (this._runs = new Client_10.Runs(this._options)));
    }
    get steps() {
        var _a;
        return ((_a = this._steps) !== null && _a !== void 0 ? _a : (this._steps = new Client_11.Steps(this._options)));
    }
    get tag() {
        var _a;
        return ((_a = this._tag) !== null && _a !== void 0 ? _a : (this._tag = new Client_12.Tag(this._options)));
    }
    get voice() {
        var _a;
        return ((_a = this._voice) !== null && _a !== void 0 ? _a : (this._voice = new Client_13.Voice(this._options)));
    }
    get templates() {
        var _a;
        return ((_a = this._templates) !== null && _a !== void 0 ? _a : (this._templates = new Client_14.Templates(this._options)));
    }
}
exports.LettaClient = LettaClient;
