// @letta-ai/letta-client/dist/serialization/resources/templates/types/TemplatesCreateAgentsResponseAgentsItemToolsItem.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsResponseAgentsItemToolsItem = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsResponseAgentsItemToolsItemToolType_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolsItemToolType");
const TemplatesCreateAgentsResponseAgentsItemToolsItemDescription_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolsItemDescription");
const TemplatesCreateAgentsResponseAgentsItemToolsItemSourceType_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolsItemSourceType");
const TemplatesCreateAgentsResponseAgentsItemToolsItemOrganizationId_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolsItemOrganizationId");
const TemplatesCreateAgentsResponseAgentsItemToolsItemName_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolsItemName");
const TemplatesCreateAgentsResponseAgentsItemToolsItemSourceCode_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolsItemSourceCode");
const TemplatesCreateAgentsResponseAgentsItemToolsItemJsonSchema_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolsItemJsonSchema");
const TemplatesCreateAgentsResponseAgentsItemToolsItemArgsJsonSchema_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolsItemArgsJsonSchema");
const TemplatesCreateAgentsResponseAgentsItemToolsItemCreatedById_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolsItemCreatedById");
const TemplatesCreateAgentsResponseAgentsItemToolsItemLastUpdatedById_1 = require("./TemplatesCreateAgentsResponseAgentsItemToolsItemLastUpdatedById");
exports.TemplatesCreateAgentsResponseAgentsItemToolsItem = core.serialization.object({
    id: core.serialization.string().optional(),
    toolType: core.serialization.property("tool_type", TemplatesCreateAgentsResponseAgentsItemToolsItemToolType_1.TemplatesCreateAgentsResponseAgentsItemToolsItemToolType.optional()),
    description: TemplatesCreateAgentsResponseAgentsItemToolsItemDescription_1.TemplatesCreateAgentsResponseAgentsItemToolsItemDescription.optional(),
    sourceType: core.serialization.property("source_type", TemplatesCreateAgentsResponseAgentsItemToolsItemSourceType_1.TemplatesCreateAgentsResponseAgentsItemToolsItemSourceType.optional()),
    organizationId: core.serialization.property("organization_id", TemplatesCreateAgentsResponseAgentsItemToolsItemOrganizationId_1.TemplatesCreateAgentsResponseAgentsItemToolsItemOrganizationId.optional()),
    name: TemplatesCreateAgentsResponseAgentsItemToolsItemName_1.TemplatesCreateAgentsResponseAgentsItemToolsItemName.optional(),
    tags: core.serialization.list(core.serialization.string()).optional(),
    sourceCode: core.serialization.property("source_code", TemplatesCreateAgentsResponseAgentsItemToolsItemSourceCode_1.TemplatesCreateAgentsResponseAgentsItemToolsItemSourceCode.optional()),
    jsonSchema: core.serialization.property("json_schema", TemplatesCreateAgentsResponseAgentsItemToolsItemJsonSchema_1.TemplatesCreateAgentsResponseAgentsItemToolsItemJsonSchema.optional()),
    argsJsonSchema: core.serialization.property("args_json_schema", TemplatesCreateAgentsResponseAgentsItemToolsItemArgsJsonSchema_1.TemplatesCreateAgentsResponseAgentsItemToolsItemArgsJsonSchema.optional()),
    returnCharLimit: core.serialization.property("return_char_limit", core.serialization.number().optional()),
    createdById: core.serialization.property("created_by_id", TemplatesCreateAgentsResponseAgentsItemToolsItemCreatedById_1.TemplatesCreateAgentsResponseAgentsItemToolsItemCreatedById.optional()),
    lastUpdatedById: core.serialization.property("last_updated_by_id", TemplatesCreateAgentsResponseAgentsItemToolsItemLastUpdatedById_1.TemplatesCreateAgentsResponseAgentsItemToolsItemLastUpdatedById.optional()),
});
