// @letta-ai/letta-client/dist/serialization/resources/templates/types/TemplatesCreateAgentsResponseAgentsItemEmbeddingConfig.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsResponseAgentsItemEmbeddingConfig = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingEndpointType_1 = require("./TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingEndpointType");
const TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingEndpoint_1 = require("./TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingEndpoint");
const TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingChunkSize_1 = require("./TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingChunkSize");
const TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigHandle_1 = require("./TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigHandle");
const TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureEndpoint_1 = require("./TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureEndpoint");
const TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureVersion_1 = require("./TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureVersion");
const TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureDeployment_1 = require("./TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureDeployment");
exports.TemplatesCreateAgentsResponseAgentsItemEmbeddingConfig = core.serialization.object({
    embeddingEndpointType: core.serialization.property("embedding_endpoint_type", TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingEndpointType_1.TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingEndpointType),
    embeddingEndpoint: core.serialization.property("embedding_endpoint", TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingEndpoint_1.TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingEndpoint.optional()),
    embeddingModel: core.serialization.property("embedding_model", core.serialization.string()),
    embeddingDim: core.serialization.property("embedding_dim", core.serialization.number()),
    embeddingChunkSize: core.serialization.property("embedding_chunk_size", TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingChunkSize_1.TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigEmbeddingChunkSize.optional()),
    handle: TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigHandle_1.TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigHandle.optional(),
    azureEndpoint: core.serialization.property("azure_endpoint", TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureEndpoint_1.TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureEndpoint.optional()),
    azureVersion: core.serialization.property("azure_version", TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureVersion_1.TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureVersion.optional()),
    azureDeployment: core.serialization.property("azure_deployment", TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureDeployment_1.TemplatesCreateAgentsResponseAgentsItemEmbeddingConfigAzureDeployment.optional()),
});
